lib/kubeclient/common.rb

```diff
@@ -78,11 +92,48 @@ def handle_exception
       raise KubeException.new(e.http_code, err_message, e.response)
     end
 
+    def discover
+      @entities = {}
+      handle_exception do
+        response = rest_client.get(@headers)
+        result = JSON.parse(response)
+        result['resources'].each do |resource|
+          # mitigation for https://github.com/kubernetes/kubernetes/issues/22413:
+          next if resource['name'].include?('/') || resource['kind'] == 'ReplicationControllerDummy'
+          entity = ClientMixin.parse_definition(resource['kind'], resource['name'])
+          @entities[entity.method_names[0]] = entity if entity
+        end
+        define_entity_methods
+        @discovered = true
```

Comment:
@moolitayer why all these lines are within the `handle_exception` block? Shouldn't just the `rest_client` request be within the `handle_exception` block?